### ISSUE TITLE
Support blockDeviceMappings with instance store NVMe volume to provisioned nodes

### ISSUE THREAD

**Issue by maciej-ciach** [aws/karpenter#2372]
## Version

**Karpenter**: v0.16.0

**Kubernetes**: v1.20.0

## Expected Behavior
I would like to use EC2 instances with instance store volumes instead of provisioning additional EBS devices

## Actual Behavior
I tried to configure provisioner using example of blockDeviceMapping from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/block-device-mapping-concepts.html#instance-block-device-mapping 
```
    amiFamily: Bottlerocket
    blockDeviceMappings:
      - deviceName: /dev/xvdb
        virtualName: ephemeral0
```
But I'm getting provisioner errors when scheduling pods.
## Steps to Reproduce the Problem

Configure provisioner with blockDeviceMappings block above

**Comment by FernandoMiguel** [comment]
@maciej-ciach have you seen https://github.com/aws/karpenter/blob/074ed02/examples/provisioner/multiple-ebs.yaml ?

**Comment by bwagner5** [comment]
This would be a good thing to add! Probably not very difficult either if anyone would like to take a stab at a PR for it

**Comment by FernandoMiguel** [comment]
@bwagner5 add to where?
the code i linked is already in this repo

**Comment by bwagner5** [comment]
`virtualName` is not implemented yet and would be required for non-ebs instance storage.

**Comment by TechDawg270** [comment]
i will try to prep a PR for this over the next 2-3 days

**Comment by saurabhpal97** [comment]
Tried to fix in https://github.com/aws/karpenter/pull/2554

**Comment by d-m** [comment]
Is it possible to modify instance store volumes via `blockDeviceMappings` now? I see the PR for this was closed but not merged.

**Comment by cep21** [comment]
Are there workarounds or examples of how I can use the on host SSD for my ephemeral storage?  Even if I need to make specific Provisioners, an example would be helpful

**Comment by bwagner5** [comment]
There's an open issue for this that has a little more activity, going to close this one as a duplicate and will track work on this front in https://github.com/aws/karpenter/issues/2723.

There is on-going work in the eks optimized AL2 AMI to setup a RAID-0 out of instance storage disks and remount containerd and kubelet. Once that PR is merged into the EKS Optimized AMI, we can then set the bootstrap flag within Karpenter to enable the new functionality and adjust the node ephemeral-storage capacity to assume that we'll use a RAID-0 setup for instance types with NVMe instance storage.  https://github.com/awslabs/amazon-eks-ami/pull/1171
